### PR TITLE
bookmark validation: move urldecode before the check of relativepath

### PR DESCRIPTION
**Pull request by ansyral**
@vwxyzh @superyyrrzz @hellosnow @qinezh @chenkennt

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Common/ValidateBookmark.cs [hunk 1]
```diff
@@ -39,12 +39,12 @@ public override void Handle(HtmlDocument document, ManifestItem manifestItem, st
             // RFC 3986: relative-ref = relative-part [ "?" query ] [ "#" fragment ]
             _linksWithBookmark[outputFile] =
                 (from node in GetNodesWithAttribute(document, "href")
-                 let link = node.GetAttributeValue("href", null)
+                 let link = HttpUtility.UrlDecode(node.GetAttributeValue("href", null))
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Common/ValidateBookmark.cs]: >link [](start = 21, length = 4)

What if file name contains `#`?

Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Common/ValidateBookmark.cs]: nice catch!

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by hellosnow** [comment]
:shipit:

**Comment by qinezh** [comment]
:shipit:

**Pull request merged**